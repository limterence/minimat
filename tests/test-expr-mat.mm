/* test-expr-mat: matrix assignment, subselect, definition */
constant matrix x = [-99.0;-9.9;];      /* matrix constant definition */
int main() {
  matrix a;
  matrix b;
  matrix c;
  sequence u;
  sequence v;
  a = [  1.1,  2.1,  3.1,  4.1; 
         5.2,  6.2,  7.2,  8.2; ];      /* construct matrix literal */
  b = [ 11.1, 12.1, 13.1, 14.1; 
        15.2, 16.2, 17.2, 18.2; ];
  printmat(a);
  printmat([;]);                        /* empty matrix */
  printmat([a; b;]);                    /* horzcat matrix */
  printmat([a', b';]);                  /* vertcat matrix */
  printmat([a, [;];]);                  /* one matrix can be empty */
  printmat([[;], b;]);
  printmat([a; [;];]);
  printmat([[;]; b;]);
  printmat(a[1, 2]);                    /* subselect matrix */
  u = 1::1;
  v = 1:2:3;
  printmat(a[u, v]);
  b[1, 3:-2:1] = a[0, v];               /* subassign matrix */
  printmat(b);
  b = a;                                /* assignment (copies values) */
  b[0,2::3] = x';
  printmat(a);
  printmat(b);
  printmat(a[1,2::3] = b[0,2::3] = [-99.9, -88.8;]);  /* chained assignment */
  printmat(a);
  printmat(b);
  printmat(a[1,0] = b[0,1] = [-77.7;]);
  printmat(a);
  printmat(b);
}
