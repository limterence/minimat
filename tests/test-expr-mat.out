[2 x 4 float]
  1.10   2.10   3.10   4.10 
  5.20   6.20   7.20   8.20 
[0 x 0 float]
[4 x 4 float]
  1.10   2.10   3.10   4.10 
  5.20   6.20   7.20   8.20 
 11.10  12.10  13.10  14.10 
 15.20  16.20  17.20  18.20 
[4 x 4 float]
  1.10   5.20  11.10  15.20 
  2.10   6.20  12.10  16.20 
  3.10   7.20  13.10  17.20 
  4.10   8.20  14.10  18.20 
[2 x 4 float]
  1.10   2.10   3.10   4.10 
  5.20   6.20   7.20   8.20 
[2 x 4 float]
 11.10  12.10  13.10  14.10 
 15.20  16.20  17.20  18.20 
[2 x 4 float]
  1.10   2.10   3.10   4.10 
  5.20   6.20   7.20   8.20 
[2 x 4 float]
 11.10  12.10  13.10  14.10 
 15.20  16.20  17.20  18.20 
[1 x 1 float]
  7.20 
[1 x 2 float]
  6.20   8.20 
[2 x 4 float]
 11.10  12.10  13.10  14.10 
 15.20   4.10  17.20   2.10 
[2 x 4 float]
  1.10   2.10   3.10   4.10 
  5.20   6.20   7.20   8.20 
[2 x 4 float]
  1.10   2.10 -99.00  -9.90 
  5.20   6.20   7.20   8.20 
[1 x 2 float]
-99.90 -88.80 
[2 x 4 float]
  1.10   2.10   3.10   4.10 
  5.20   6.20 -99.90 -88.80 
[2 x 4 float]
  1.10   2.10 -99.90 -88.80 
  5.20   6.20   7.20   8.20 
[1 x 1 float]
-77.70 
[2 x 4 float]
  1.10   2.10   3.10   4.10 
-77.70   6.20 -99.90 -88.80 
[2 x 4 float]
  1.10 -77.70 -99.90 -88.80 
  5.20   6.20   7.20   8.20 
